src/OrleansProviders/Streams/Common/Monitors/IObjectPoolMonitor.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Monitor track block pool related metrics
+    /// </summary>
+    public interface IObjectPoolMonitor
+    {
+        #region Event driven metrics
+        /// <summary>
+        /// Track every time cache allocate memory from block pool
+        /// </summary>
+        /// <param name="blockCount"></param>
+        void TrackObjectAllocatedByCache(int blockCount);
```

Comment:
This monitor is related to IObjectPool, not the cache.  IObjectPool only allocates one object at a time, so a count here doesn't really make sense.  I don't think it's a question of exposing implementation details as much as providing monitoring appropriate for the class being monitored.

Along the same line of reasoning, TrackObjectAllocated is more appropriate.  We should not be making assumptions about who is requesting the allocation.